src/pages/NotFound.tsx

```diff
@@ -1,15 +1,48 @@
 import React from "react";
 import { useNavigate } from "react-router-dom";
+import Button from "../components/ui/Button";
+import { ButtonSize } from "../utils/types";
+import { useGetProfileQuery } from "../features/user/apiSlice";
+import { getJWT } from "../utils/helper";
 
 const NotFound = () => {
   const navigate = useNavigate();
+  const jwt = getJWT();
+  const {data: profile} = useGetProfileQuery(jwt);
+  const role = profile?.role;
+
+  
+  const navigateBasedOnRole = (role: string) => { 
+    switch (role) {
+      case "Admin":
+        navigate("/forms");
+        break;
+      case "Coach":
+        navigate("/overview");
+        break;
+      case "Applicant" || "Prospect":
+        navigate("/home");
+        break;
+      default:
+        navigate("/login");
+        break;
+    }
+  }
```

Comment:
Since we just want to go back to where we were, I think we don't need to know the role of the user. we can just go back using `navigate(-1)`